crates/nu-cmd-lang/src/core_commands/overlay/list.rs

```diff
@@ -24,27 +25,57 @@ impl Command for OverlayList {
 
     fn run(
         &self,
-        _engine_state: &EngineState,
+        engine_state: &EngineState,
         stack: &mut Stack,
         call: &Call,
         _input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        let active_overlays_engine: Vec<Value> = stack
-            .active_overlays
-            .iter()
-            .map(|s| Value::string(s, call.head))
-            .collect();
+        let show_hidden = call.has_flag(engine_state, stack, "hidden")?;
 
-        Ok(Value::list(active_overlays_engine, call.head).into_pipeline_data())
+        let overlays: Vec<Value> = if show_hidden {
+            // Show all overlays from engine state (including hidden ones)
+            engine_state
+                .scope
+                .overlays
+                .iter()
+                .filter_map(|(name, _)| {
+                    let overlay_name = String::from_utf8_lossy(name);
+                    stack
+                        .active_overlays
+                        .iter()
+                        .all(|name| name.as_str() != overlay_name) // filter already existing overlays
+                        .then_some(Value::string(overlay_name, call.head))
```

Comment:
Is `--hidden` supposed to show all overlays, or just hidden overlays? The comment above implies all, but this section filters out the active overlays

Maybe instead `overlay list` should output `table<name: string, active: bool>`? Then we don't need a flag, and you could get hidden overlays with `overlay list | where active`